src/core/lib/security/transport/security_connector.c

```diff
@@ -509,7 +556,8 @@ static grpc_error *ssl_check_peer(grpc_security_connector *sc,
 static void ssl_channel_check_peer(grpc_exec_ctx *exec_ctx,
```

Comment:
It looks to me that here, in the case of LB, you should do the secure naming check on the `targets_info[target_uri]`. Same applies to `check_call_host`.